test/gui/shared/steps/server_context.py

```diff
@@ -120,7 +122,11 @@ def step(context, user, folder_name):
 
 @Given('user "|any|" has uploaded file with content "|any|" to "|any|" in the server')
 def step(context, user, file_content, file_name):
-    webdav.create_file(user, file_name, file_content)
+    if get_config('predefined_users'):
+        user = get_username_for_user(user)
+        webdav.upload_file(user, file_name)
+    else:
+        webdav.create_file(user, file_name, file_content)
```

Comment:
this doesn't feel right. let create a temp file in a temp location and do similar to `upload_file` in `create_file` function if predefined users.
```suggestion
    webdav.create_file(user, file_name, file_content)
```